cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResources.java

```diff
@@ -110,6 +110,13 @@ public CreatedDiskEncryptionSet createDiskEncryptionSet(DiskEncryptionSetCreatio
             String sourceVaultId = String.format("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.KeyVault/vaults/%s",
                     azureClient.getCurrentSubscription().subscriptionId(), vaultResourceGroupName, vaultName);
 
+            // Check for the existence of keyVault user has specified before creating disk encryption set(DES).
+            // If keyVault is wrong or user lacks permissions to access it, granting access permissions to this keyVault for DES would fail.
+            if (!azureClient.checkKeyVaultExistence(vaultResourceGroupName, vaultName)) {
+                throw new IllegalArgumentException(String.format("vaultName \"%s\" either does not exists or user does not have permissions to access it. " +
+                                "Kindly check if the vault & encryption key exists and correct encryption key url is specified.",
```

Comment:
Nits:
* `vaultName` -> `Vault with name` (not sure the customer would understood `vaultName` as it is an internal identifier extracted from the key URL)
* `url` -> `URL`